fluentd/run.sh

```diff
@@ -129,32 +125,68 @@ else
 fi
 export K8S_FILTER_REMOVE_KEYS
 
-if [ "${USE_MUX:-}" = "true" ] ; then
-    TOTAL_MEMORY_LIMIT=`echo $MUX_MEMORY_LIMIT |  sed -e "s/[Kk]/*1024/g;s/[Mm]/*1024*1024/g;s/[Gg]/*1024*1024*1024/g;s/i//g" | bc`
+# How many outputs?
+if [ "${USE_MUX_CLIENT:-}" = "true" ] ; then
+    # mux client has just one output.
+    DIV=1
 else
-    TOTAL_MEMORY_LIMIT=`echo $FLUENTD_MEMORY_LIMIT |  sed -e "s/[Kk]/*1024/g;s/[Mm]/*1024*1024/g;s/[Gg]/*1024*1024*1024/g;s/i//g" | bc`
-fi
-BUFFER_SIZE_LIMIT=`echo $BUFFER_SIZE_LIMIT |  sed -e "s/[Kk]/*1024/g;s/[Mm]/*1024*1024/g;s/[Gg]/*1024*1024*1024/g;s/i//g" | bc`
-if [ $BUFFER_SIZE_LIMIT -eq 0 ]; then
-    BUFFER_SIZE_LIMIT=1048576
+    # fluend usually has 2 putputs.
+    DIV=2
+    if [ "$ES_COPY" = "true" ]; then
+        DIV=`expr $DIV \* 2`
+    fi
 fi
 
-DIV=1
-if [ "$ES_HOST" != "$OPS_HOST" ] || [ "$ES_PORT" != "$OPS_PORT" ] ; then
-    # using ops cluster
-    DIV=`expr $DIV \* 2`
+FILE_BUFFER_PATH=/var/lib/fluentd/buffer
+if [ ! -d $FILE_BUFFER_PATH ]; then
+    mkdir -p $FILE_BUFFER_PATH
 fi
+
 if [ -n "${MUX_CLIENT_MODE:-}" -o "${USE_MUX_CLIENT:-}" = "true" ] ; then
     DIV=`expr $DIV \* 2`
 fi
+# Get the available disk size; use 1/4 of it
+DF_LIMIT=$(df -B1 $FILE_BUFFER_PATH | grep -v Filesystem | awk '{print $2}')
+DF_LIMIT=${DF_LIMIT:-0}
+DF_LIMIT=$(expr $DF_LIMIT / 4) || :
+if [ $DF_LIMIT -eq 0 ]; then
+    echo "ERROR: No disk space is available for file buffer in $FILE_BUFFER_PATH."
+    exit 1
+fi
+# Given buffer limit per output
+TOTAL_LIMIT=$(echo ${FILE_BUFFER_LIMIT:-2Gi} | sed -e "s/[Kk]/*1024/g;s/[Mm]/*1024*1024/g;s/[Gg]/*1024*1024*1024/g;s/i//g" | bc) || :
+# In total
+TOTAL_LIMIT=$(expr $TOTAL_LIMIT \* $DIV) || :
+if [ $TOTAL_LIMIT -le 0 ]; then
+    echo "ERROR: Invalid file buffer limit ($FILE_BUFFER_LIMIT) is given.  Failed to convert to bytes."
+    exit 1
+fi
+# Available disk space is less than the specified size.
+if [ $DF_LIMIT -lt $TOTAL_LIMIT ]; then
+    echo "WARNING: Available disk space ($DF_LIMIT bytes) is less than the user specified file buffer limit ($FILE_BUFFER_LIMIT)."
+    TOTAL_LIMIT=$DF_LIMIT
+fi
+
+if [ -z $TOTAL_LIMIT -o $TOTAL_LIMIT -eq 0 ]; then
+    if [ "$USE_MUX" = "true" ]; then
+        # 75 % of 2G
+        TOTAL_LIMIT=1610612736
```

Comment:
Is this making the assumption that this default size is small enough that we don't care how much space is taken up on the volume?  How do we sure this value won't fill up the disk on which the buffer is located?